src/components/stateTracker/js/StateTracker.js

```diff
@@ -0,0 +1,100 @@
+                funcName: "fluid.stateTracker.startTracking",
+                args: ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"]
+                                 // changeEvaluator, changeListener, polling interval
+            },
+            stopTracking: {
+                funcName: "fluid.stateTracker.stopTracking",
+                args: ["{that}", "{arguments}.0", "{arguments}.1"]
+                                 // changeListener, intervalID
+            },
+            monitorChange: {
+                funcName: "fluid.stateTracker.monitorChange",
+                args: ["{that}", "{arguments}.0"]
+                                 // monitor info structure
+            },
+            initMonitorInfo: {
+                funcName: "fluid.stateTracker.initMonitorInfo",
+                args: ["{arguments}.0"]
+                       // changeEvaluator object or function
+            }
+        }
+    });
+
+    // Initiate polling.
+    // @param changeEvaluator - an object used to evaluate if the state
+    //                          has changed.  It must have an evaluateChange()
+    //                          function.
+    // @param changeListener  - a listener to handle the onStateChange event.
+    // @param interval        - optional delay between calls to check the state's
+    //                          current value (msec).
+    // @return                - the intervalID.
+    fluid.stateTracker.startTracking = function (that, changeEvaluator, changeListener, interval) {
+        var monitor = that.initMonitorInfo(changeEvaluator);
+        that.events.onStateChange.addListener(changeListener);
+        if (!!interval) {
+            that.interval = interval;
+        }
+        monitor.intervalID = setInterval(function () {
+            that.monitorChange(monitor);
+        }, that.interval);
+        return monitor.intervalID;
+    };
+
+    // Stop polling.
+    // @param changeListener - the listener to remove.
+    // @param intervalID     - the interval to clear.
+    fluid.stateTracker.stopTracking = function (that, listener, intervalID) {
+        that.events.onStateChange.removeListener(listener);
+        clearInterval(intervalID);
+    };
+
+    // Callback to pass to setInterval() to periodically check a state.
+    fluid.stateTracker.monitorChange = function (that, monitor) {
+        if (monitor.changeEvaluator.evaluateChange()) {
+            that.events.onStateChange.fire(monitor.changeEvaluator);
+        }
+    };
+
+    // Initialize a monitor object for passing to a function that periodically
+    // checks for chanages in the state of a process.
+    // @param changeEvaluator - an object to evaluate if the state has changed
+    //                          that has a function named evaluateChange().
+    //                          Provided by client.
+    // @return                - the monitor object.
+    fluid.stateTracker.initMonitorInfo = function (changeEvaluator) {
```

Comment:
A clearer pattern would have the changeEvaluator defined as an invoker on the component itself, with the initial value "fluid.notImplemented" in order to signal that the user needs to override to make the component usable - see https://issues.fluidproject.org/browse/FLUID-5733 for description